### ISSUE TITLE
ArmNN fails to build when linking libpipeCommon.a

### ISSUE THREAD

**Issue by the-promised-LAN** [ARM-software/armnn#463]
Hello!

**TLDR:**
When trying to build ARM NN with either TF Lite or ONNX support using Android NDK r20b, process fails when creating libpipeCommon.a

**Details:**
Trying to create Docker images for building ArmNN with either ONNX or TensorFlow Lite support (separate images) with Android NDK **r20b** support, following the [official guides](https://developer.arm.com/solutions/machine-learning-on-arm/developer-material/how-to-guides/configuring-the-arm-nn-sdk-build-environment-for-tensorflow-lite/single-page). Following the 20.08 SW stack versions, in both ONNX/TF Lite, ArmNN build fails on the first task of `make`, that is, building the libpipeCommon.a from the [profiling/common/src](https://github.com/ARM-software/armnn/tree/branches/armnn_20_08/profiling/common/src) directory, stating `No such file or directory`, see crash message below:
```
-- The C compiler identification is Clang 8.0.7
-- The CXX compiler identification is Clang 8.0.7
-- Check for working C compiler: /home/armnn-devenv/android-ndk-r20b/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android29-clang
-- Check for working C compiler: /home/armnn-devenv/android-ndk-r20b/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android29-clang -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /home/armnn-devenv/android-ndk-r20b/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android29-clang++
-- Check for working CXX compiler: /home/armnn-devenv/android-ndk-r20b/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android29-clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Finding Boost
-- Boost version: 1.64.0
-- Found the following Boost libraries:
--   unit_test_framework
--   filesystem
--   system
--   program_options
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - found
-- Found Threads: TRUE  
-- Flatbuffers headers are located at: /home/armnn-devenv/flatbuffers/include
-- Flatbuffers library located at: /home/armnn-devenv/flatbuffers/lib/libflatbuffers.a
-- Tf Lite generated header found at: /home/armnn-devenv/tensorflow/tensorflow/lite/schema
-- Looking for valgrind/memcheck.h
-- Looking for valgrind/memcheck.h - not found
-- Caffe parser support is disabled
-- Tensorflow parser support is disabled
-- Armnn Serializer support is disabled
-- ArmNN Quantizer support is disabled
-- PyArmNN wheel package is disabled
-- PyArmNN source package is disabled
-- Including backend common library into the build: /home/armnn-devenv/armnn/src/backends/backendsCommon/common.cmake
-- Including backend common library into the build: /home/armnn-devenv/armnn/src/backends/aclCommon/common.cmake
-- Including backend into the build: /home/armnn-devenv/armnn/src/backends/neon/backend.cmake
-- Including backend into the build: /home/armnn-devenv/armnn/src/backends/reference/backend.cmake
-- Including backend into the build: /home/armnn-devenv/armnn/src/backends/cl/backend.cmake
-- Adding object library dependency to armnn: armnnBackendsCommon
-- Adding object library dependency to armnn: armnnAclCommon
-- Adding object library dependency to armnn: armnnNeonBackend
-- Adding object library dependency to armnn: armnnNeonBackendWorkloads
-- Adding object library dependency to armnn: armnnRefBackend
-- Adding object library dependency to armnn: armnnRefBackendWorkloads
-- Adding object library dependency to armnn: armnnClBackend
-- Adding object library dependency to armnn: armnnClBackendWorkloads
-- Adding object library dependency to UnitTests: armnnBackendsCommonUnitTests
-- Adding object library dependency to UnitTests: armnnAclCommonUnitTests
-- Adding object library dependency to UnitTests: armnnNeonBackendUnitTests
-- Adding object library dependency to UnitTests: armnnRefBackendUnitTests
-- Adding object library dependency to UnitTests: armnnClBackendUnitTests
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    PROTOBUF_ROOT
    TF_GENERATED_SOURCES

-- Build files have been written to: /home/armnn-devenv/armnn/build
Scanning dependencies of target pipeCommon
[  0%] Building CXX object profiling/common/src/CMakeFiles/pipeCommon.dir/CommandHandlerFunctor.cpp.o
[  0%] Building CXX object profiling/common/src/CMakeFiles/pipeCommon.dir/CommandHandlerKey.cpp.o
[  0%] Building CXX object profiling/common/src/CMakeFiles/pipeCommon.dir/CommandHandlerRegistry.cpp.o
[  0%] Building CXX object profiling/common/src/CMakeFiles/pipeCommon.dir/CommonProfilingUtils.cpp.o
[  0%] Building CXX object profiling/common/src/CMakeFiles/pipeCommon.dir/NetworkSockets.cpp.o
[  0%] Building CXX object profiling/common/src/CMakeFiles/pipeCommon.dir/PacketVersionResolver.cpp.o
[  1%] Building CXX object profiling/common/src/CMakeFiles/pipeCommon.dir/SwTrace.cpp.o
[  1%] Linking CXX static library libpipeCommon.a
Error running link command: No such file or directory
make[2]: *** [profiling/common/src/libpipeCommon.a] Error 2
profiling/common/src/CMakeFiles/pipeCommon.dir/build.make:250: recipe for target 'profiling/common/src/libpipeCommon.a' failed
CMakeFiles/Makefile2:2056: recipe for target 'profiling/common/src/CMakeFiles/pipeCommon.dir/all' failed
make[1]: *** [profiling/common/src/CMakeFiles/pipeCommon.dir/all] Error 2
make: *** [all] Error 2
Makefile:116: recipe for target 'all' failed
```
Interesting thing is that this exact error comes up whenever trying to build either ONNX or TF Lite (both using Android NDK **r20b**), however, nothing comes up if building with Caffe support and Android NDK **r17b**.

I attached the Dockerfile with the ticket (just remove .txt extension to run).
[Dockerfile.txt](https://github.com/ARM-software/armnn/files/5330879/Dockerfile.txt)

Any help would be greatly appreciated!

**Comment by MikeJKelly** [comment]
Hi @the-promised-LAN 

we have dockerfiles at https://github.com/ARM-software/armnn/blob/branches/armnn_20_08/docker/ that might provide help but it looks like you're not building the basePipeServer, can you try to build armnn with BUILD_BASE_PIPE_SERVER=1?

Best regards
Mike

**Comment by the-promised-LAN** [comment]
Hi @MikeJKelly,

Thanks for the reply! Adding BUILD_BASE_PIPE_SERVER=1 unfortunately did not fix the issue; same error pops up. Using Android NDK r17b instead of r20b fixed the issue (with both ONNX and TF Lite). 

Before closing this ticket, is there a way to check which file is `make` looking for when raising an error? I think it's closely related to the differences between NDK r17b and r20b, potentially due to switch from creating a standalone toolchain in r17b, and default ones provided in r20b.
